backend/src/federation/index.ts

```diff
@@ -112,8 +157,7 @@ export const startDHT = async (
         socket.on('open', function () {
           // noiseSocket fully open with the other peer
           // console.log("writing to socket");
-          socket.write(Buffer.from(`${nodeRand}`))
-          socket.write(Buffer.from(JSON.stringify(nodeAPI)))
+          socket.write(Buffer.from(JSON.stringify(apiList)))
```

Comment:
consider:
`foreach api do send(json{api.v, api.u} + \n)`

Reply:
as mentioned above, sending each api with a separate socket.write will cause several asynchron and parallel socket.on invocations, which will then have to be synchronized for writing to the same database table.

So i prefere the solution writing the whole api-list per one socket.write invocation.

Reply:
The unsyncronized thing will happen with multiple communities anyway.
I do not believe thats a problem tho.

I agree that we keep it like this for now.